cirq/google/optimizers/convert_to_sycamore_gates.py

```diff
@@ -51,12 +51,18 @@ def __init__(self,
         """
         Args:
             tabulation: If set, a tabulation for the Sycamore gate to use for
-                decomposing gates.
+                decomposing Matrix gates. If unset, an analytic calculation is
+                used for Matrix gates. To get a GateTabulation, provide its
+                constructor a base gate (in this case, usually cirq.google.SYC)
+                and a maximmum infidelity.
             ignore_failures: If set, gates that fail to convert are forwarded
                 unchanged. If not set, conversion failures raise a TypeError.
         """
         super().__init__()
         self.ignore_failures = ignore_failures
+        assert tabulation is None or isinstance(
+            tabulation, GateTabulation
+        ), "provided tabulation must be of type GateTabulation"
```

Comment:
one additional python thing: it's more idiomatic to do

```python
if [whatever check is violated]:
  raise ValueError("message")
```

instead of using asserts. A violated assertion returns an `AssertionError` whereas a `ValueError` is more appropriate

Reply:
Ah good point, done.